### DISCUSSION TITLE
mpv为什么只推荐解码线程在16个以内

### DISCUSSION THREAD

**Discussion by redomCL** [hooke007/mpv_PlayKit#420] [mpv 通用]
5900X有24个线程，不过在mpv里设置24线程解码之后日志里会有警告
![QQ截图20240227170036](https://github.com/hooke007/MPV_lazy/assets/22269254/73acdc9d-b5a4-495e-abaa-1f5f0e1e3578)
请教一下mpv为什么会只推荐16个以内的线程进行解码呢？

**Comment by hooke007** [comment]
多了也是浪费，4-8个线程照样不影响观看用的解码速度，除非搞很逆天的编码

**Comment by redomCL** [comment]
> 多了也是浪费，4-8个线程照样不影响观看用的解码速度，除非搞很逆天的编码

目前有没有mpv基于gpu-next的，从源滤镜到分离器-解码器-渲染器等完整的视频解码流程图，看了几个不太全的，非常感谢！

**Reply by hooke007** [reply]
目前哪里有流程图？gpunext也是一个vo，换的就是gpu的位置

**Comment by redomCL** [comment]
> 目前哪里有流程图？gpunext也是一个vo，换的就是gpu的位置

我现在的疑惑主要集中在osd2里面几次抗色带和encode记录，为什么会有好几次呢？晚上回头截图请教一下你

**Reply by hooke007** [reply]
page2展示的是vo-pass，gpunext没有公开文档记录过，要弄清楚这你得问开发者

**Reply by redomCL** [reply]
![2](https://github.com/hooke007/MPV_lazy/assets/22269254/fb2c275c-ed1f-4e37-9f2f-463b939c9f1c)
你好大佬！我下班回家辣，截图给你看一下，目前以我的认知范围内，实在没理解为什么会出现这种多次抗色带，然后反复的decode，encode，如果可以的话，还请劳烦一下，解释一下我的迷惑

**Reply by redomCL** [reply]
我的理解不应该是从分离器出来进行decode，然后交给渲染器做好各种预期内的抗色带，色度、分辨率计算等等后，输出到屏幕吗？为什么会多次encode

**Reply by hooke007** [reply]
前面写的是overlay，所以我认为这是OSD的渲染层
